File: testsuite/tests/typing-layouts-void/stubs.c
#include <caml/mlvalues.h>
#include <assert.h>

void void_to_void() {
}

value void_to_seven() {
  return Val_long(7);
}

void seven_to_void(value seven) {
  assert(Long_val(seven) == 7);
}

value void_to_void_bytecode(value unit) {
  assert(Long_val(unit) == 0);
  return Val_unit;
}

value void_to_seven_bytecode(value unit) {
  assert(Long_val(unit) == 0);
  return Val_long(7);
}

value seven_to_void_bytecode(value seven) {
  assert(Long_val(seven) == 7);
  return Val_unit;
}

value six_to_seven(value six) {
  assert(Long_val(six) == 6);
  return Val_long(7);
}

File: testsuite/tests/typing-layouts-void/stubs.js
//Provides: void_to_void
function void_to_void() {
}

//Provides: void_to_seven
function void_to_seven() {
  return 7;
}

//Provides: seven_to_void
function seven_to_void(seven) {
  if (seven !== 7) throw new Error("Assertion failed: seven !== 7");
}

//Provides: six_to_seven
function six_to_seven(six) {
  if (six !== 6) throw new Error("Assertion failed: six !== 6");
  return 7;
}

//Provides: six_to_seven_to_eight
function six_to_seven_to_eight(six, seven) {
  if (six !== 6) throw new Error("Assertion failed: six !== 6");
  if (seven !== 7) throw new Error("Assertion failed: seven !== 7");
  return 8;
}
